docs/spec.md

```diff
@@ -4648,10 +4648,10 @@ More formally, for all `update_index` from the index space of `updates[0]`:
       `index_vector_dim` index, if `index_vector_dim` <
       `rank(scatter_indices)`.
   * `[scatter_indices[update_scatter_index]]` otherwise.
-* For `do` in `axes(inputs[0])`,
-  * `full_start_index[do]` = `start_index[ds]` if
-      `do = scatter_dims_to_operand_dims[ds]`.
-  * `full_start_index[do]` = `0` otherwise.
+* For `d_operand` in `axes(inputs[0])`,
+  * `full_start_index[d_operand]` = `start_index[d_start]` if
+      `d_operand = scatter_dims_to_operand_dims[d_start]`.
+  * `full_start_index[d_operand]` = `0` otherwise.
```

Comment:
We're also planning to follow up on addressing the out of bounds behavior in the near future, similar to GatherOp, right?

Reply:
Yes and the behavior of HLO is to skip computing out of bound indices: https://github.com/tensorflow/tensorflow/blob/master/tensorflow/compiler/xla/hlo/evaluator/hlo_evaluator.cc#L3020-L3025